pkg/plugins/manifest.go

```diff
@@ -82,49 +84,130 @@ func readPluginManifest(body []byte) (*pluginManifest, error) {
 	return manifest, nil
 }
 
+func getChksums(dpath, manifestPath string, log log.Logger) (map[string]string, error) {
+	manifestPath = filepath.Clean(manifestPath)
+
+	chksums := map[string]string{}
+	if err := filepath.Walk(dpath, func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if fi.IsDir() {
+			return nil
+		}
+
+		path = filepath.Clean(path)
+
+		// Handle symbolic links
+		if fi.Mode()&os.ModeSymlink == os.ModeSymlink {
+			finalPath, err := filepath.EvalSymlinks(path)
+			if err != nil {
+				return err
+			}
+
+			log.Debug("Handling symlink", "source", path, "target", finalPath)
+
+			info, err := os.Stat(finalPath)
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+
+			if _, err := filepath.Rel(dpath, finalPath); err != nil {
+				return fmt.Errorf("symbolic link %q targets a file outside of the plugin directory: %q", path, finalPath)
+			}
+
+			if finalPath == manifestPath {
+				return nil
+			}
+
+			fi = info
+		}
+
+		if path == manifestPath {
+			return nil
+		}
+
+		// Ignore images
+		for _, sfx := range []string{"png", "gif", "svg"} {
+			if strings.HasSuffix(fi.Name(), fmt.Sprintf(".%s", sfx)) {
+				return nil
+			}
+		}
+
+		h := sha256.New()
+		f, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		if _, err := io.Copy(h, f); err != nil {
+			return err
+		}
+
+		relPath, err := filepath.Rel(dpath, path)
+		if err != nil {
+			return err
+		}
+		chksums[relPath] = fmt.Sprintf("%x", h.Sum(nil))
```

Comment:
Curious why appending nil? Maybe a comment?

Reply:
@kylebrandt `h.Sum(nil)` is the standard way of getting the checksum.

Reply:
A bit weird, but that's Go :)